packages/kbn-ftr-common-functional-services/services/saml_auth/get_auth_provider.ts

```diff
@@ -28,26 +20,10 @@ export interface AuthProvider {
 
 export interface AuthProviderProps {
   config: Config;
-  kibanaServer: KibanaServer;
-  log: ToolingLog;
 }
 
-export const getAuthProvider = async (props: AuthProviderProps) => {
-  const { config, log, kibanaServer } = props;
+export const getAuthProvider = (props: AuthProviderProps) => {
+  const { config } = props;
   const isServerless = !!props.config.get('serverless');
-  if (isServerless) {
-    return new ServerlessAuthProvider(config);
-  }
-
-  const provider = new StatefulAuthProvider();
-  // TODO: Move it to @kbn-es package, so that roles and its mapping are created before FTR services loading starts.
-  // 'viewer' and 'editor' roles are available by default, but we have to create 'admin' role
-  const adminRoleMapping = JSON.parse(
-    fs.readFileSync(require.resolve(STATEFUL_ADMIN_ROLE_MAPPING_PATH), 'utf8')
-  );
-  await createRole({ roleName: 'admin', roleMapping: adminRoleMapping, kibanaServer, log });
-  const roles = Object.keys(provider.getSupportedRoleDescriptors());
-  // Creating roles mapping for mock-idp
-  await createRoleMapping({ name: MOCK_IDP_REALM_NAME, roles, config, log });
```

Comment:
Instead of 2 API calls in FTR service we copy `roles.yml` to ES config directory and create role mapping via API but using template